docs/sharding.md

```diff
@@ -0,0 +1,79 @@
+---
+title: Sharding
+type: docs
+menu: thanos
+slug: /Sharding.md
+---
+
+# Store and Compactor Sharding
+
+Store and Compactor Sharding is for Long Term Retention Storage.
```

Comment:
This feels a bit weird, not sure what you are trying to say here. Maybe we don't need this?